Sprint 2

daikoziyevBaha opened the pull request:
Sprint 2

daikoziyevBaha committed:
Typescript, eslint, tslint added
Login, register, profile pages refactored using Block component
Validation funcs for input added

daikoziyevBaha committed:
ChatOpenPage, 404, 500 pages refactored using Block class
added Stylelint

mikebars1995 commented on src/components/ChatDetails/ChatDetails.scss:
```diff
@@ -53,10 +55,11 @@
 .message-left {
   max-width: 43%;
   background: #F8F8F8;
```
**Можно лучше**

все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде

mikebars1995 commented on src/index.html:
```diff
@@ -7,6 +7,6 @@
   </head>
   <body>
     <div id="app"></div>
```
**Нужно исправить**


По семантике нужно помещать весь главный контент страницы в тег `main`.


Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

mikebars1995 commented on src/index.html:
```diff
@@ -7,6 +7,6 @@
   </head>
```
**Можно лучше**


Лучше установить корректный язык страницы `<html lang="ru">`

Раз интерфейс проекта русскоязычный, то и декларация языка должна ему соответствовать. Некоторые устройства могут постоянно предлагать перевести сайт на другой язык, или ничего не спрашивая, тихо менять страницу. Вот одна из таких историй: https://habr.com/ru/post/650907/

mikebars1995 commented on src/pages/ProfileEditPage/ProfileEditPage.ts:
```diff
@@ -0,0 +1,244 @@
+        input: new Input({
+            attr: {
+                class: 'straight-label__value',
+                name: 'last_name',
```
**Нужно исправить**


По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ

mikebars1995 commented on src/components/Block/Block.ts:
```diff
@@ -0,0 +1,237 @@
+    [key: string]: any
+}
+
+class Block {
```
**Можно лучше**

лучше всего передавать тип пропсов в виде дженерика в `Block`, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Block`. Будет очень универсально 

```js
abstract class Block<Props extends Record<string, any> = unknown> {

```

mikebars1995 commented on src/pages/ProfileEditPage/ProfileEditPage.ts:
```diff
@@ -0,0 +1,244 @@
+                    event.preventDefault();
+                    event.stopPropagation();
+                    if (!validateEmail(event.target.value)) {
+                        emailInput.setProps({ error: 'Неверная почта' });
```
**Нужно исправить**

На странице редактировании профиля не показываются сообщения валидации

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Добавлен `ESLint` и `Stylelint`
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что моковые данные вынесены в отдельный файл

Что нужно исправить:

* По семантике нужно помещать весь главный контент страницы в тег `main`.
* По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ 
* На странице редактировании профиля не показываются сообщения валидации

Что можно улучшить:

* все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде
* Лучше установить корректный язык страницы `<html lang="ru">`
* лучше всего передавать тип пропсов в виде дженерика в `Block`, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Block`. Будет очень универсально 



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.

daikoziyevBaha committed:
bugFix
ProfileEditPage error error text
app lang
class Block type
Main layout for full page

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
